app/decorators/table_sorters/hearings_sorter.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module TableSorters
+  class HearingsSorter
+    def initialize(hearings, sort_order)
+      @hearings = hearings
+      @sort_order = sort_order
+    end
+
+    def sorted_hearings
+      order_by_asc_or_desc(@hearings.sort_by { |h| h.hearing_days.map(&:to_datetime) })
+    end
+
+    def sorted_hearing(hearing)
+      if @sort_order.eql?('date_desc')
+        hearing.hearing_days.map(&:to_datetime).sort.reverse
+      else
+        hearing.hearing_days.map(&:to_datetime).sort
+      end
+    end
```

Comment:
If I understand correctly, the reversed option here will only happen when sorting by date, so it will never occur in either of the child classes `HearingsProviderSorter` or `HearingsTypeSorter`. It might be worth having third class `HearingsDateSorter` (used on line 34 of `app/decorators/prosecution_case_decorator.rb`) something like:

```ruby
module TableSorter
  class HearingsSorter

    ...

    def sorter_hearing(hearing)
      hearing.hearing_day.map(&:to_datetime).sort
    end

    ...

  end
end

module TableSorter
  class HearingsDateSorter << HearingsSorter
    def sorter_hearing(hearing)
      order_by_asc_or_desc(super)
    end
  end
end
```

Reply:
@jrmhaig I have created  a new HearingsDateSorter class which subclasses HearingsSorter and implements sorted_hearing as you suggest.